subsys/bluetooth/audio/shell/bap.c

```diff
@@ -361,10 +330,19 @@ static void lc3_audio_timer_timeout(struct k_work *work)
 		}
 		lc3_sdu_cnt++;
 		seq_num++;
-	}
+	} while (lc3_sdu_cnt <= prime_count);
 }
 
-static K_WORK_DELAYABLE_DEFINE(audio_send_work, lc3_audio_timer_timeout);
+static K_WORK_DELAYABLE_DEFINE(audio_send_work, lc3_audio_send_data);
+
+void sdu_sent_cb(struct bt_bap_stream *stream)
+{
+	if (txing_stream == NULL || txing_stream->qos == NULL) {
+		return;
+	}
+
+	k_work_schedule(&audio_send_work, K_MSEC(0));
```

Comment:
Hmm, I don't think this will work as intended. 

A single call to `lc3_audio_send_data` will call `bt_bap_stream_send` multiple times, which in turn will trigger this callback multiple times. 

So for example if `lc3_audio_send_data` calls `bt_bap_stream_send` twice, then

1 x call to `lc3_audio_send_data` calls `bt_bap_stream_send` x 2, which triggers `sdu_sent_cb` x 2, which calls `lc3_audio_send_data` x 2, which calls `bt_bap_stream_send` x 4, etc. 

IMO either `lc3_audio_send_data` needs to only TX once, or `sdu_sent_cb` should only call `lc3_audio_send_data` every `prime_count`, or am I missing something?

Reply:
you have a point on this being called twice the first time - but the do { .. } while (lc3_sdu_cnt <= prime_count) should bail out after one iteration on subsequent runs as the lc3_sdu_cnt should be well above prime_count next time, so should not accumulate.  I can add a bool to check if we are in 'priming' mode and skip the call in sdu_sent_cb .. or maybe just skip the 'prime_count-1' first runs in the callback (will cost an extra compare, but probably ok).  Any preference?

Reply:
Aha, that makes sense. 

What about this approach? 

In `cmd_start_sine` instead of just doing `k_work_schedule(&audio_send_work, K_MSEC(0));`, we do

```
const size_t prime_count = 2U;

for (size_t i = 0U; i < prime_count; i++) {
    lc3_audio_send_data(...)
}
```

thus removing the prime count from the `lc3_audio_send_data` function all together, so a single call means a single TX. This fixes the above "issue" and makes the function slightly simpler as well (similar approach used by the BAP Broadcast Source sample (https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/broadcast_audio_source/src/main.c)).

Reply:
yeah, looks more clean.  I'll try that